src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -124,103 +126,99 @@ def validate_metadata(
         else None
     )
 
+    metadata = parse_metadata()
     validation = metadata.is_valid_for(
         requirements.lockfile_hex_digest,
         interpreter_constraints,
         python_setup.interpreter_universe,
         req_strings,
     )
-
     if validation:
         return
 
-    def tool_message_parts(
-        requirements: (ToolCustomLockfile | ToolDefaultLockfile),
-    ) -> Iterator[str]:
-
-        tool_name = requirements.options_scope_name
-        uses_source_plugins = requirements.uses_source_plugins
-        uses_project_interpreter_constraints = requirements.uses_project_interpreter_constraints
-
-        yield "You are using "
+    message = (
+        "".join(
+            _invalid_tool_lockfile_error(
+                validation,
+                requirements,
+                actual_interpreter_constraints=interpreter_constraints,
+                lockfile_interpreter_constraints=metadata.valid_for_interpreter_constraints,
+            )
+        ).strip()
+        if isinstance(requirements, (ToolCustomLockfile, ToolDefaultLockfile))
+        else str(validation.failure_reasons)
+    )
 
-        if isinstance(requirements, ToolDefaultLockfile):
-            yield "the `<default>` lockfile provided by Pants "
-        elif isinstance(requirements, ToolCustomLockfile):
-            yield f"the lockfile at {requirements.file_path} "
+    if python_setup.invalid_lockfile_behavior == InvalidLockfileBehavior.error:
+        raise InvalidLockfileError(message)
+    logger.warning("%s", message)
+
+
+def _invalid_tool_lockfile_error(
+    validation: LockfileMetadataValidation,
+    requirements: ToolCustomLockfile | ToolDefaultLockfile,
+    *,
+    actual_interpreter_constraints: InterpreterConstraints,
+    lockfile_interpreter_constraints: InterpreterConstraints,
+) -> Iterator[str]:
+    tool_name = requirements.options_scope_name
+    uses_source_plugins = requirements.uses_source_plugins
+    uses_project_interpreter_constraints = requirements.uses_project_interpreter_constraints
+
+    yield "You are using "
```

Comment:
No changes to the content of this function. Only whitespace changes + using ternary expressions where possible rather than distinct if/else blocks, per our style guide: https://www.pantsbuild.org/docs/style-guide#prefer-conditional-expressions-ternary-expressions